pkg/cmd/cli/cmd/set/probe.go

```diff
@@ -0,0 +1,409 @@
+)
+
+type ProbeOptions struct {
+	Out io.Writer
+	Err io.Writer
+
+	Filenames         []string
+	ContainerSelector string
+	Selector          string
+	All               bool
+
+	Builder *resource.Builder
+	Infos   []*resource.Info
+
+	Encoder runtime.Encoder
+
+	ShortOutput bool
+	Mapper      meta.RESTMapper
+
+	PrintObject            func(runtime.Object) error
+	UpdatePodSpecForObject func(runtime.Object, func(spec *kapi.PodSpec) error) (bool, error)
+
+	Readiness bool
+	Liveness  bool
+	Unset     bool
+
+	OpenTCPSocket string
+	HTTPGet       string
+	Command       []string
+
+	FlagSet       func(string) bool
+	HTTPGetAction *kapi.HTTPGetAction
+
+	// Length of time before health checking is activated.  In seconds.
+	InitialDelaySeconds *int
+	// Length of time before health checking times out.  In seconds.
+	TimeoutSeconds *int
+	// How often (in seconds) to perform the probe.
+	PeriodSeconds *int
+	// Minimum consecutive successes for the probe to be considered successful after having failed.
+	// Must be 1 for liveness.
+	SuccessThreshold *int
+	// Minimum consecutive failures for the probe to be considered failed after having succeeded.
+	FailureThreshold *int
+}
+
+// NewCmdProbe implements the set probe command
+func NewCmdProbe(fullName string, f *clientcmd.Factory, out, errOut io.Writer) *cobra.Command {
+	options := &ProbeOptions{
+		Out: out,
+		Err: errOut,
+
+		ContainerSelector: "*",
+	}
+	cmd := &cobra.Command{
+		Use:     "probe RESOURCE/NAME [--readiness|--liveness] ...",
+		Short:   "Update a probe on a pod template",
+		Long:    probeLong,
+		Example: fmt.Sprintf(probeExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(options.Complete(f, cmd, args))
+			kcmdutil.CheckErr(options.Validate())
+			if err := options.Run(); err != nil {
+				// TODO: move met to kcmdutil
+				if err == cmdutil.ErrExit {
+					os.Exit(1)
+				}
+				kcmdutil.CheckErr(err)
+			}
+		},
+	}
+
+	kcmdutil.AddPrinterFlags(cmd)
+	cmd.Flags().StringVarP(&options.ContainerSelector, "containers", "c", options.ContainerSelector, "The names of containers in the selected pod templates to change - may use wildcards")
+	cmd.Flags().StringVarP(&options.Selector, "selector", "l", options.Selector, "Selector (label query) to filter on")
+	cmd.Flags().BoolVar(&options.All, "all", options.All, "Select all resources in the namespace of the specified resource types")
+	cmd.Flags().StringSliceVarP(&options.Filenames, "filename", "f", options.Filenames, "Filename, directory, or URL to file to use to edit the resource.")
+
+	cmd.Flags().BoolVar(&options.Unset, "unset", options.Unset, "If true, remove the specified probe(s).")
+	cmd.Flags().BoolVar(&options.Readiness, "readiness", options.Readiness, "Set a readiness probe to indicate when this container should receive traffic")
+	cmd.Flags().BoolVar(&options.Liveness, "liveness", options.Liveness, "Set a liveness probe to verify this container is running")
```

Comment:
I have --clear-all in triggers.  I think I'm going to go with remove and
remove-all since volume uses it.

On Thu, Feb 18, 2016 at 4:57 PM, Fabiano Franz <notifications@github.com>
wrote:

> In pkg/cmd/cli/cmd/set/probe.go
> <https://github.com/openshift/origin/pull/7331#discussion_r53389187>:
>
> > +					os.Exit(1)
> > +				}
> > +				kcmdutil.CheckErr(err)
> > +			}
> > +		},
> > +	}
> > +
> > +	kcmdutil.AddPrinterFlags(cmd)
> > +	cmd.Flags().StringVarP(&options.ContainerSelector, "containers", "c", options.ContainerSelector, "The names of containers in the selected pod templates to change - may use wildcards")
> > +	cmd.Flags().StringVarP(&options.Selector, "selector", "l", options.Selector, "Selector (label query) to filter on")
> > +	cmd.Flags().BoolVar(&options.All, "all", options.All, "Select all resources in the namespace of the specified resource types")
> > +	cmd.Flags().StringSliceVarP(&options.Filenames, "filename", "f", options.Filenames, "Filename, directory, or URL to file to use to edit the resource.")
> > +
> > +	cmd.Flags().BoolVar(&options.Unset, "unset", options.Unset, "If true, remove the specified probe(s).")
> > +	cmd.Flags().BoolVar(&options.Readiness, "readiness", options.Readiness, "Set a readiness probe to indicate when this container should receive traffic")
> > +	cmd.Flags().BoolVar(&options.Liveness, "liveness", options.Liveness, "Set a liveness probe to verify this container is running")
>
> oc set probe dc/registry --unset --readyness
> oc set probe dc/registry --remove --readyness
> oc set probe dc/registry --clear --readyness
>
> I'd say --clear seems more fluid, but I'm also ok with --remove for
> consistency. Feels odd to say set --unset.
>
> BTW, should oc set probe dc/registry --clear remove both (clear all)?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/7331/files#r53389187>.
>